### ISSUE TITLE
Support for flex and width/height/margin with percentage

### ISSUE THREAD

**Issue by BogdanArdelean** [tokkozhin/react-native-neomorph-shadows#25]
Hello!

While developing our app, we found the need to use flex and width/height/margin with percentage along with the Neomorph component.

Since the Neomorph component doesn't support those attributes out of the box, we found a rather hackish HoC solution that may fit our needs. It's still a proof of concept and it needs to be thoroughly tested and extended, but the idea remains the same.

The approach is to enclose the Neomorph component into a View and exploit the onLayout callback to only render it after the layout finished, setting the according width and height.

A snippet of code that exemplifies the approach:

```javascript

export function withNeomorphFlex(Component) {
    return class WithNeomorphFlex extends React.Component {
        constructor(props) {
            super(props);
           
            const {style} = props;
            this.state = {
                neomorphStyle: {
                    shadowOpacity: (style && style.shadowOpacity),
                    shadowRadius: (style && style.shadowRadius),
                    borderRadius: (style && style.borderRadius),
                    backgroundColor: (style && style.backgroundColor), // transparent does not work on iOS
                    width: 0,
                    height: 0,
                    position: 'absolute',
                },
                layoutFinished: false,
            }
        }

        _onLayout = ({nativeEvent}) => {
            const width = nativeEvent.layout.width;
            const height = nativeEvent.layout.height;
            this.setState({
                neomorphStyle: {
                    ...this.state.neomorphStyle,
                    width: width,
                    height: height,
                },
                layoutFinished: true,
            })
        };

        render() {
            const style = {
                ...this.props.style,
                backgroundColor: TRANSPARENT,
            };

            return (
                <View onLayout={this._onLayout} style={style}>
                    {this.state.layoutFinished &&
                    <Neomorph
                        // other props
                        style={this.state.neomorphStyle}
                    />
                    }
                    <Component
                        {...this.props}
                        style={{
                            ...style,
                            marginTop: 0,
                            marginLeft: 0,
                            marginRight: 0,
                            marginBottom: 0,
                            margin: 0,

                            paddingTop: 0,
                            paddingLeft: 0,
                            paddingRight: 0,
                            paddingBottom: 0,
                            padding: 0,

                            width: '100%',
                            height: '100%'
                        }}
                    >
                        {this.props.children}
                    </Component>
                </View>
            )
        }
    }
}
```

There are some inherent drawbacks with this approach in a sense that the Neomorph shadows will appear only after the first rendered frame, but I think that many of us can live with that. 
If most of the screen is static and it is paired with a transition, the flicker is barely noticeable.

If you think this is a good enough approach, we could perfect this technique and maybe include it as an experimental feature.

Would like to hear your thoughts on this.

**Comment by tokkozhin** [comment]
Yes, as I wrote in one of last commits about this feature, it might be worth making an experimental component. Thx for code, I'll take your with some changes. Thx for idea! Wait for the next version!

**Comment by tokkozhin** [comment]
Implemented in a new v1.1.0! 🎉  
You can use experimental **ShadowFlex**, **NeomorphFlex**.
